src/platform/Linux/ThreadStackManagerImpl.cpp

```diff
@@ -685,15 +685,169 @@ CHIP_ERROR ThreadStackManagerImpl::_WriteThreadNetworkDiagnosticAttributeToTlv(A
     case ThreadNetworkDiagnostics::Attributes::RouteTableList::Id:
     case ThreadNetworkDiagnostics::Attributes::SecurityPolicy::Id:
     case ThreadNetworkDiagnostics::Attributes::OperationalDatasetComponents::Id:
```

Comment:
Why is OperationalDatasetComponents being encoded as a list?  Pre-existing, so separate PR is probably OK.

Reply:
https://github.com/project-chip/connectedhomeip/issues/18906